Guilherme Yoshikawa da Silva

RA: 818133945

Curso: Ciência da Computação

Aula 03 - Exercício Prático 3

## 1. Faça um Programa que peça dois números e imprima o maior deles.

In [4]:
a = int(input('Insira um número: '))
b = int(input('Insira um número: '))

if a > b:
    print('Maior número: {}'.format(a))
else:
    print('Maior número: {}'.format(b))

Insira um número:  5
Insira um número:  3


Maior número: 5


## 2. Faça um Programa que verifique se uma letra digitada é vogal ou consoante.

In [12]:
vogal = ['a', 'e', 'i', 'o', 'u']

a = input('Insira uma letra: ')

if a == 'a' or a == 'e' or a == 'i' or a == 'o' or a == 'u':
    print('A palavra "{}" uma vogal.'.format(a))
else:
    print('A palavra é "{}" uma consoante.'.format(a))

Insira uma letra:  b


A palavra é "b" uma consoante.


## 3. Faça um programa para a leitura de duas notas parciais de um aluno. O programa deve calcular a média alcançada por aluno e apresentar:
- A mensagem &quot;Aprovado&quot;, se a média alcançada for maior ou igual a sete;
- A mensagem &quot;Reprovado&quot;, se a média for menor do que sete;
- A mensagem &quot;Aprovado com Distinção&quot;, se a média for igual a dez.

In [23]:
nota1 = float(input('Insira a primeira nota: '))
nota2 = float(input('Insira a segunda nota: '))

media = ((nota1 + nota2)/2)

if media >= 7.0:
    print('Aprovado!')
elif media == 10.0:
    print('Aprovado com Distinção!')
else:
    print('Reprovado!')

Insira a primeira nota:  10
Insira a segunda nota:  9


Aprovado!


## 4. Faça um Programa que leia três números e mostre-os em ordem decrescente.

In [14]:
num1 = int(input('Número 1: '))
num2 = int(input('Número 2: '))
num3 = int(input('Número 3: '))

if num1 > num2 > num3:
    print(num1, num2, num3)
if num1 > num3 > num2:
    print(num1, num3, num2)
if num2 > num1 > num3:
    print(num2, num1, num3)
if num2 > num3 > num1:
    print(num2, num3, num1)
if num3 > num1 > num2:
    print(num3, num1, num2)
if num3 > num2 > num1:
    print(num3, num2, num1)
if num1 == num2 == num3:
    print(num1, num2, num3)


Número 1:  1
Número 2:  5
Número 3:  2


5 2 1


## 5. As Organizações Tabajara resolveram dar um aumento de salário aos seus colaboradores e lhe contrataram para desenvolver o programa que calculará os reajustes. Faça um programa que recebe o salário de um colaborador e o reajuste segundo o seguinte critério, baseado no salário atual:

- salários até R$ 280,00 (incluindo) : aumento de 20%

- salários entre R$ 280,00 e R$ 700,00 : aumento de 15%

- salários entre R$ 700,00 e R$ 1500,00 : aumento de 10%

- salários de R$ 1500,00 em diante : aumento de 5% Após o aumento ser realizado, informe na tela:

- o salário antes do reajuste;

- o percentual de aumento aplicado;

- o valor do aumento;

- o novo salário, após o aumento.

In [6]:
salario = float(input('O valor do salário: R$ '))

if salario <= 280:
    aumento = ((20/100) * salario)
    reajuste = (salario + aumento)
    print('O percentual de aumento aplicado: 20%')
    print('O valor do aumento: R$ {}'.format(aumento))
    print('O novo salário, após o aumento: R$ {}'.format(reajuste))
if salario >= 280 <= 700:
    aumento = ((30/100) * salario)
    reajuste = (salario + aumento)
    print('O percentual de aumento aplicado: 15%')
    print('O valor do aumento: {}'.format(aumento))
    print('O novo salário, após o aumento: R$ {}'.format(reajuste))
    
if salario >= 700 <= 1500:
    aumento = ((30/100) * salario)
    reajuste = (salario + aumento)
    print('O percentual de aumento aplicado: 10%')
    print('O valor do aumento: {}'.format(aumento))
    print('O novo salário, após o aumento: R$ {}'.format(reajuste))
    
if salario > 1500:
    aumento = ((30/100) * salario)
    reajuste = (salario + aumento)
    print('O percentual de aumento aplicado: 5%')
    print('O valor do aumento: {}'.format(aumento))
    print('O novo salário, após o aumento: R$ {}'.format(reajuste))

O valor do salário: R$  500


O percentual de aumento aplicado: 15%
O valor do aumento: 150.0
O novo salário, após o aumento: R$ 650.0


## 6. Faça um Programa que leia um número e exiba o dia correspondente da semana. (1-Domingo, 2- Segunda, etc.), se digitar outro valor deve aparecer valor inválido.

In [3]:
num = int(input('Insira um número: '))

if num == 1:
    print('1 - Domingo')
if num == 2:
    print('2 - Segunda')
if num == 3:
    print('3 - Terça')
if num == 4:
    print('4 - Quarta')
if num == 5:
    print('5 - Quinta')
if num == 6:
    print('6 - Sexta')
if num == 7:
    print('7 - Sabádo')
if num > 7:
    print('Valor inválido')

Insira um número:  9


Valor inválido


## 7. Faça um programa que lê as duas notas parciais obtidas por um aluno numa disciplina ao longo de um semestre, e calcule a sua média. A atribuição de conceitos obedece à tabela abaixo: Média de Aproveitamento Conceito

Entre 9.0 e 10.0 A

Entre 7.5 e 9.0 B

Entre 6.0 e 7.5 C

Entre 4.0 e 6.0 D

Entre 4.0 e zero E

O algoritmo deve mostrar na tela as notas, a média, o conceito correspondente e a mensagem “APROVADO” se o conceito for A, B ou C ou “REPROVADO” se o conceito for D ou E.

In [6]:
nota1 = float(input('Primeira nota: '))
nota2 = float(input('Segunda nota: '))

media = ((nota1 + nota2) / 2)
print('Média: {}'.format(media))

if media >= 9 and media <= 10:
    status = 'A'
elif media >= 7.5 and media < 9:
    status = 'B'
elif media >= 6 and media < 7.5:
    status = 'C'
elif media >= 4 and media < 6:
    status = 'D'
elif media < 4:
    status = 'E'
    
print('Status: ', status)

if status == "A" or status == "B" or status == "C":
    print("APROVADO")
else:
    print("REPROVADO")

Primeira nota:  10
Segunda nota:  10


Média: 10.0
Status:  A
APROVADO


## 8. Faça um Programa que peça os 3 lados de um triângulo. O programa deverá informar se os valores podem ser um triângulo. Indique, caso os lados formem um triângulo, se o mesmo é: equilátero, isósceles ou escaleno. Dicas:

- Três lados formam um triângulo quando a soma de quaisquer dois lados for maior que o terceiro;

- Triângulo Equilátero: três lados iguais;

- Triângulo Isósceles: quaisquer dois lados iguais;

- Triângulo Escaleno: três lados diferentes;

In [2]:
l1 = int(input('Insira o lado do triângulo: '))
l2 = int(input('Insira outro lado do triângulo: '))
l3 = int(input('Insira o última lado do triângulo: '))

if l1 > (l2 + l3) or l2 > (l1 + l3) or l3 > (l1 + l2):
    print('Não pode ser um triangulo')
elif l1 == l2 == l3:
    print('Triâgulo Equilatero')
elif l1 == l2 or l1 == l3 or l2 == l3:
    print('Triâgulo Isósceles')
else:
    print('Triâgulo Escaleno')

Insira o lado do triângulo:  1
Insira outro lado do triângulo:  0
Insira o última lado do triângulo:  0


Não pode ser um triangulo


## 9. Faça um programa que calcule as raízes de uma equação do segundo grau, na forma ax2 + bx + c. O programa deverá pedir os valores de a, b e c e fazer as consistências, informando ao usuário nas seguintes situações:

- Se o usuário informar o valor de A igual a zero, a equação não é do segundo grau e o programa não deve fazer pedir os demais valores, sendo encerrado;

- Se o delta calculado for negativo, a equação não possui raizes reais. Informe ao usuário e encerre o programa;

- Se o delta calculado for igual a zero a equação possui apenas uma raiz real; informe-a ao usuário;

- Se o delta for positivo, a equação possui duas raiz reais; informe-as ao usuário;

In [10]:
import math

a = int(input('Insira o valor de A: ' ))

if(a == 0):
    print('A equação não é do segundo grau.')
else:
    b = int(input('Insira o valor de B: ' ))
    c = int(input('Insira o valor de C: ' ))
    delta = b * b - (4 * a * c)

if delta < 0:
    print('A equação não possui raízes reais.')
elif delta == 0:
    raiz = -b / (2 * a)
    print('Delta = 0 , raiz real = ', raiz)
else:
    r1 = (-b + math.sqrt(delta) ) / (2 * a)
    r2 = (-b - math.sqrt(delta) ) / (2 * a)
    print('Raizes: ', r1, ' e ', r2)

Insira o valor de A:  1
Insira o valor de B:  0
Insira o valor de C:  1


A equação não possui raízes reais.


## 10. Faça um Programa para um caixa eletrônico. O programa deverá perguntar ao usuário a valor do saque e depois informar quantas notas de cada valor serão fornecidas. As notas disponíveis serão as de 1, 5, 10, 50 e 100 reais. O valor mínimo é de 10 reais e o máximo de 600 reais. O programa não deve se preocupar com a quantidade de notas existentes na máquina.

- Exemplo 1: Para sacar a quantia de 256 reais, o programa fornece duas notas de 100, uma nota de 50, uma nota de 5 e uma nota de 1;

- Exemplo 2: Para sacar a quantia de 399 reais, o programa fornece três notas de 100, uma nota de 50, quatro notas de 10, uma nota de 5 e quatro notas de 1.

In [12]:
saque = int(input('Valor de saque: R$ '))

nota100 = int(saque / 100)
saque = saque % 100
    
nota50 = int(saque/50)
saque = saque % 50

nota10 = int(saque/10)
saque = saque % 10

nota5 = int(saque/5)
saque = saque % 5

nota1 = saque
    
print('Notas R$100,00 = ', nota100)
print('Notas R$ 50,00 = ', nota50)
print('Notas R$ 10,00 = ', nota10)
print('Notas R$  5,00 = ', nota5)
print('R$  1,00 = ', nota1)

Valor de saque: R$  256


Notas R$100,00 =  2
Notas R$ 50,00 =  1
Notas R$ 10,00 =  0
Notas R$  5,00 =  1
R$  1,00 =  1


## 11. Faça um programa que faça 5 perguntas para uma pessoa sobre um crime. As perguntas são:

- Telefonou para a vítima?

- Esteve no local do crime?

- Mora perto da vítima?

- Devia para a vítima?

- Já trabalhou com a vítima?

O programa deve no final emitir uma classificação sobre a participação da pessoa no crime. Se a pessoa responder positivamente a 2 questões ela deve ser classificada como Suspeita, entre 3 e 4 como Cúmplice e 5 como Assassino. Caso contrário,ele será classificado como Inocente.

In [16]:
pontos = 0
perguntas = [
    input('Telefonou para a vítima? (sim/não): '),
    input('Esteve no local do crime? (sim/não): '),
    input('Mora perto da vítima? (sim/não): '),
    input('Devia para a vítima? (sim/não): '),
    input('Já trabalhou com a vítima? (sim/não): ')
]

for perguntas in perguntas:
    if perguntas == 'sim':
        pontos += 1
        
if pontos == 2:
    print('Suspeita')
elif pontos >= 3 and pontos <= 4:
    print('Cúmplice')
elif pontos == 5:
    print('Assassino')
else:
    print('Inocente')


Telefonou para a vítima? (sim/não):  sim
Esteve no local do crime? (sim/não):  não
Mora perto da vítima? (sim/não):  não
Devia para a vítima? (sim/não):  sim
Já trabalhou com a vítima? (sim/não):  não


Suspeita


## 12. Uma fruteira está vendendo frutas com a seguinte tabela de preços: Até 5 Kg Acima de 5 Kg

- Morango R$ 2,50 por Kg R$ 2,20 por Kg
- Maçã R$ 1,80 por Kg R$ 1,50 por Kg

Se o cliente comprar mais de 8 Kg em frutas ou o valor total da compra ultrapassar R$ 25,00, receberá ainda um desconto de 10% sobre este total. Escreva um algoritmo para ler a quantidade (em Kg) de morangos e a quantidade (em Kg) de maças adquiridas e escreva o valor a ser pago pelo cliente.

In [18]:
preco = 0
morango = float(input("Insira a quantidade de Kg de morangos: "))
maca = float(input("Insira a quantidade de Kg de maçãs: "))

if morango < 5:
    preco += morango * 2.50
else:
    preco += morango * 2.20
    
if maca < 5:
    preco += maca * 1.80
else:
    preco += maca * 1.50
    
if preco > 25.00:
    preco *= 0.90

print('Valor a ser pago: R$ ', preco)

Insira a quantidade de Kg de morangos:  120
Insira a quantidade de Kg de maçãs:  20


Valor a ser pago: R$  264.6


## Loops

## 13. Faça um programa que peça uma nota, entre zero e dez. Mostre uma mensagem caso o valor seja inválido e continue pedindo até que o usuário informe um valor válido.

In [19]:
nota = int(input("Insira uma nota entre zero e dez: "))
while nota < 0 or nota > 10:
    print("Valor inválido.")
    nota = int(input("Insira uma nota entre zero e dez: "))
else:
    print("Valor válido.")

Insira uma nora entre zero e dez:  199


Valor inválido.


Insira uma nota entre zero e dez:  1


Valor válido.


## 14. Faça um programa que leia e valide as seguintes informações: 

- Nome: maior que 3 caracteres;

- Idade: entre 0 e 150;

- Salário: maior que zero;

- Sexo: f ou m;

- Estado Civil: s, c, v, d;

In [ ]:
nome = input("Insira o nome: ")
while len(nome) <= 3:
    nome = input("Insira o nome: ")
    
idade = int(input("Insira a idade: "))
while idade < 0 or idade > 150:
    idade = int(input("Insira a idade: "))
    
salario = int(input("Insira o salário: "))
while salario <= 0:
    salario = int(input("Insira o salário: "))
    
sexo = input("Insira o sexo: (f / m) ")
while not (sexo in ['f', 'm']):
    sexo = input("Insira o sexo (f / m): ")
    
estado_civil = input("Insira o estado civil: (s / c / v / d)")
while not (estado_civil in ['s', 'c', 'v', 'd']):
    estado_civil = input("Insira o estado civil: (s / c / v / d)")

Insira o nome:  guilherme


## 15. Faça um programa que leia 5 números e informe o maior número.

In [1]:
maior_num = int(input("Insira o primeiro número: "))
for numero in range(2,6):
    z = int(input(f"Insira o {numero} número: "))
    if z > maior_num:
        maior_num = z
print("Maior número digitado: ", maior_num)

Insira o primeiro número:  10
Insira o 2 número:  20
Insira o 3 número:  20
Insira o 4 número:  10
Insira o 5 número:  50


Maior número digitado:  50


## 16. Faça um programa que imprima na tela apenas os números ímpares entre 1 e 50.

In [8]:
for impar in range(1, 51):
    if impar % 2 != 0:
        print(impar)


1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49


## 17. Desenvolva um gerador de tabuada, capaz de gerar a tabuada de qualquer número inteiro entre 1 a 10. O usuário deve informar de qual numero ele deseja ver a tabuada. A saída deve ser conforme o exemplo abaixo:
Tabuada de 5:

5 X 1 = 5

5 X 2 = 10

...

5 X 10 = 50

In [13]:
numero = int(input("Qual o número deseja ver a tabuada?"))

for gerador in range(1,11):
    tabuada = gerador * numero
    print(f"{numero} X {gerador} = {tabuada}")

Qual o número deseja ver a tabuada? 5


5 X 1 = 5
5 X 2 = 10
5 X 3 = 15
5 X 4 = 20
5 X 5 = 25
5 X 6 = 30
5 X 7 = 35
5 X 8 = 40
5 X 9 = 45
5 X 10 = 50


## 18. A série de Fibonacci é formada pela seqüência 1,1,2,3,5,8,13,21,34,55,... Faça um programa capaz de gerar a série até o n−ésimo termo.

In [18]:
nesimo = int(input("Digite o n-ésimo termo: "))

fibonacci = []
a = 0
b = 1

for i in range(nesimo):
    print(a)
    aux = b
    b = a + b
    a = aux
    

Digite o n-ésimo termo:  8


0
1
1
2
3
5
8
13


## 19. Faça um programa que calcule o fatorial de um número inteiro fornecido pelo usuário. 
Ex.: 5!=5.4.3.2.1=120

In [28]:
num = int(input("Digite um número inteiro: "))
for i in range(1, num):
    num = num * i
print(f"Fatorial =", num)

Digite um número inteiro:  5


Fatorial = 120


## 20. O Sr. Manoel Joaquim possui uma grande loja de artigos de R$ 1,99, com cerca de 10 caixas. Para agilizar o cálculo de quanto cada cliente deve pagar ele desenvolveu um tabela que contém o número de itens que o cliente comprou e ao lado o valor da conta. Desta forma a atendente do caixa precisa apenas contar quantos itens o cliente está levando e olhar na tabela de preços. Você foi contratado para desenvolver o programa que monta esta tabela de preços, que conterá os preços de 1 até 50 produtos, conforme o exemplo abaixo: 

Lojas Quase Dois - Tabela de preços

1 - R$ 1.99

2 - R$ 3.98

...

50 - R$ 99.50

In [32]:
for i in range(1, 51):
    preco = i * 1.99
    print(f"{i} - {preco}")

1 - 1.99
2 - 3.98
3 - 5.97
4 - 7.96
5 - 9.95
6 - 11.94
7 - 13.93
8 - 15.92
9 - 17.91
10 - 19.9
11 - 21.89
12 - 23.88
13 - 25.87
14 - 27.86
15 - 29.85
16 - 31.84
17 - 33.83
18 - 35.82
19 - 37.81
20 - 39.8
21 - 41.79
22 - 43.78
23 - 45.77
24 - 47.76
25 - 49.75
26 - 51.74
27 - 53.73
28 - 55.72
29 - 57.71
30 - 59.7
31 - 61.69
32 - 63.68
33 - 65.67
34 - 67.66
35 - 69.65
36 - 71.64
37 - 73.63
38 - 75.62
39 - 77.61
40 - 79.6
41 - 81.59
42 - 83.58
43 - 85.57
44 - 87.56
45 - 89.55
46 - 91.54
47 - 93.53
48 - 95.52
49 - 97.51
50 - 99.5


## 21. O Sr. Manoel Joaquim acaba de adquirir uma panificadora e pretende implantar a metodologia da tabelinha, que já é um sucesso na sua loja de 1,99. Você foi contratado para desenvolver o programa que monta a tabela de preços de pães, de 1 até 50 pães, a partir do preço do pão informado pelo usuário, conforme o exemplo abaixo:
Preço do pão: R$ 0.18

Panificadora Pão de Ontem - Tabela de preços

1 - R$ 0.18

2 - R$ 0.36

...

50 - R$ 9.00

In [34]:
pao = 0.18
quantidade = int(input("Quantos pães deseja levar?"))

for i in range(1,quantidade):
    preco = quantidade * pao
print(f"{quantidade} - R${preco}")
    

Quantos pães deseja levar? 50


50 - R$9.0


## 22. O Sr. Manoel Joaquim expandiu seus negócios para além dos negócios de 1,99 e agora possui uma loja de conveniências. Faça um programa que implemente uma caixa registradora rudimentar. O programa deverá receber um número desconhecido de valores referentes aos preços das mercadorias. Um valor zero deve ser informado pelo operador para indicar o final da compra. O programa deve então mostrar o total da compra e perguntar o valor em dinheiro que o cliente forneceu, para então calcular e mostrar o valor do troco. Após esta operação, o programa deverá voltar ao ponto inicial, para registrar a próxima compra. A saída deve ser conforme o exemplo abaixo:
Lojas Tabajara

Produto 1: R$ 2.20

Produto 2: R$ 5.80

Produto 3: R$ 0

Total: R$ 9.00

Dinheiro: R$ 20.00

Troco: R$ 11.00

...

In [3]:
total = 0
preco = -1
i = 0
while preco != 0:
    preco = float(input("Insira o preço do item (Digite '0' para finalizar a compra): "))
    total += preco
    i += 1
    if preco == 0:
        print(f"Total: R$ {total}")
        pago = float(input("Insira o valor a ser pago: "))
        print("Troco: R$ ", pago - total)
    else:
        print(f"Produto {i}: R$ {preco}")

Insira o preço do item (Digite '0' para finalizar a compra):  2.20


Produto 1: R$ 2.2


Insira o preço do item (Digite '0' para finalizar a compra):  5.80


Produto 2: R$ 5.8


Insira o preço do item (Digite '0' para finalizar a compra):  0


Total: R$ 8.0


Insira o valor a ser pago:  20


Troco: R$  12.0


## 23. Foi feita uma estatística em cinco cidades brasileiras para coletar dados sobre acidentes de trânsito. Foram obtidos os seguintes dados:

- Código da cidade;

- Número de veículos de passeio (em 1999);

- Número de acidentes de trânsito com vítimas (em 1999). Deseja-se saber:

- Qual o maior e menor índice de acidentes de transito e a que cidade pertence;

- Qual a média de veículos nas cinco cidades juntas;

- Qual a média de acidentes de trânsito nas cidades com menos de 2.000 veículos de passeio.

In [ ]:
codigo = int(input("Código da cidade: "))
veiculo = int(input("Número de veículos de passeio (em 1999):"))
acidente = int(input("Número de acidentes de trânsito com vítimas (em 1999). Deseja-se saber:"))




## 24. Faça um programa que receba o valor de uma dívida e mostre uma tabela com os seguintes dados: valor da dívida, valor dos juros, quantidade de parcelas e valor da parcela. Os juros e a quantidade de parcelas seguem a tabela abaixo:
Quantidade de Parcelas % de Juros sobre o valor inicial da dívida

1 - 0

3 - 10

6 - 15

9 - 20

12 - 25

Exemplo de saída do programa:

Valor da Dívida Valor dos Juros Quantidade de Parcelas Valor da Parcela

R$ 1.000,00 - 0 - 1 - R$ 1.000,00

R$ 1.100,00 - 100 - 3 - R$ 366,00

R$ 1.150,00 - 150 - 6 - R$ 191,67

In [19]:
divida = float(input("Valor da dívida: R$ "))
parcela = int(input("Quantidade de parcelas: "))
juros = 0
print("")
print("Dívida \t Valor Juros \t Qtde Parcelas \t Valor da Parcela")

for i in range(0, 5):
    if i == 0:
        juros == 0
    elif i == 1:
        juros == 10
    else:
        juros += 5
        
    juros_total = divida * (juros / 100)
    divida_total = divida + juros_total

    if i == 0:
        parcela = 1
    else:
        parcela = i * 3
    if parcela != 0:
        valor_parcela = divida_total / parcela
    else:
        valor_parcela = divida_total
        
    print(f"R$ {divida_total} \t {juros_total} \t {parcela} \t R$ %.2f"%(valor_parcela))

Valor da dívida: R$  1000
Quantidade de parcelas:  6



Dívida 	 Valor Juros 	 Qtde Parcelas 	 Valor da Parcela
R$ 1000.0 	 0.0 	 1 	 R$ 1000.00
R$ 1000.0 	 0.0 	 3 	 R$ 333.33
R$ 1050.0 	 50.0 	 6 	 R$ 175.00
R$ 1100.0 	 100.0 	 9 	 R$ 122.22
R$ 1150.0 	 150.0 	 12 	 R$ 95.83


# 25. Faça um programa que mostre os n termos da Série a seguir:
S = 1/1 + 2/3 + 3/5 + 4/7 + 5/9 + ... + n/m.

Imprima no final a soma da série.

In [5]:
termo = int(input('Digite o número de termos: '))
soma = "S = "
a = 0
b = 1

for n in range(1, termo + 1):
    a += n / b
    soma += str(n) + "/" + str(b)
    if n != termo:
        soma += " + "
    b += 2

print(soma)
print('Soma da série: %s' % a)

Digite o número de termos:  5


S = 1/1 + 2/3 + 3/5 + 4/7 + 5/9
Soma da série: 3.393650793650793
